traits/ctraits.c

```diff
@@ -4529,28 +4546,18 @@ static trait_setattr setattr_property_handlers[] = {
     (trait_setattr)post_setattr_trait_python, NULL};
 
 static PyObject *
-_trait_property(trait_object *trait, PyObject *args)
+_set_property(trait_object *trait, PyObject *args)
 {
     PyObject *get, *set, *validate;
     int get_n, set_n, validate_n;
 
-    if (PyTuple_GET_SIZE(args) == 0) {
-        if (trait->flags & TRAIT_PROPERTY) {
-            return PyTuple_Pack(
-                3, trait->delegate_name, trait->delegate_prefix,
-                trait->py_validate);
-        }
-        else {
-            Py_INCREF(Py_None);
-            return Py_None;
-        }
-    }
-
     if (!PyArg_ParseTuple(
             args, "OiOiOi", &get, &get_n, &set, &set_n, &validate,
             &validate_n)) {
+        PyErr_SetString(PyExc_ValueError, "Invalid arguments.");
```

Comment:
If `PyArg_ParseTuple` fails, it will already have set an appropriate exception type and an informative exception message. Is there a reason to override that with our own exception type and message?

Reply:
No good reason, we can use the exception from `PyArg_ParseTuple`